skxray/fitting/xrf_model.py

```
@@ -807,6 +816,98 @@ def setup_element_model(self, elemental_line, default_area=1e5):
+                name, line = elemental_line.split('_')
+                line = line.lower()
+                e = Element(name)
+                if 'k' in line:
+                    e_cen = e.emission_line[line]
+                elif 'l' in line:
+                    # only the first line for L
+                    e_cen = e.emission_line['la1']
+                else:
+                    # only the first line for M
+                    e_cen = e.emission_line['ma1']
+                return e_cen
+
+            e1_cen = get_line_energy(element_line1)
+            e2_cen = get_line_energy(element_line2)
+
+            # no '-' allowed in prefix name in lmfit
+            pre_name = 'pileup_' + elemental_line.replace('-', '_') + '_'
+            gauss_mod = ElementModel(prefix=pre_name)
+            gauss_mod.set_param_hint('e_offset',
+                                     value=self.compton_param['e_offset'].value,
+                                     expr='e_offset')
+            gauss_mod.set_param_hint('e_linear',
+                                     value=self.compton_param['e_linear'].value,
+                                     expr='e_linear')
+            gauss_mod.set_param_hint('e_quadratic',
+                                     value=self.compton_param['e_quadratic'].value,
+                                     expr='e_quadratic')
+            gauss_mod.set_param_hint('fwhm_offset',
+                                     value=self.compton_param['fwhm_offset'].value,
+                                     expr='fwhm_offset')
+            gauss_mod.set_param_hint('fwhm_fanoprime',
+                                     value=self.compton_param['fwhm_fanoprime'].value,
+                                     expr='fwhm_fanoprime')
+            gauss_mod.set_param_hint('epsilon', value=self.epsilon, vary=False)
+
+            area_name = pre_name + 'area'
+            if area_name in self.params:
+                default_area = self.params[area_name]['value']
```

This will silently ignore user input.